src/DIRAC/Core/Utilities/Os.py

```diff
@@ -147,6 +149,52 @@ def sourceEnv(timeout, cmdTuple, inputEnv=None):
     return result
 
 
-@deprecated("Will be removed in DIRAC 8.1", onlyOnce=True)
+@deprecated("Will be removed in DIRAC 9.0", onlyOnce=True)
 def which(executable):
     return shutil.which(executable)
+
+
+def findImage(continer_root="container/apptainer/alma9/"):
+    """Finds the image for the current platform
+    
+       This looks into location "${CVMFS_locations}/${container_root}/${platform}/"
+       and expects to find one of the following platforms:
+       - x86_64
+       - aarch64
+       - ppc64le
+
+    """
+    from DIRAC.ConfigurationSystem.Client.Helpers.Operations import Operations
+
+    plat = DIRAC.gConfig.getValue("LocalSite/Platform", platform.machine())
+    DIRAC.gLogger.info(f"Platform: {plat}")
+    
+    # NB: platform compatibility is more complex than the following simple identification. 
+    # sources: 
+    # https://stackoverflow.com/questions/45125516/possible-values-for-uname-m
+    # https://en.wikipedia.org/wiki/Uname
+
+    if plat.lower() == "amd64":
+        plat = "x86_64"
+    if plat.lower().startswith("arm"):
+        plat = "aarch64"
+    if plat.lower().startswith("ppc64"):
+        plat = "ppc64le"
+
+    if plat not in ["x86_64", "aarch64", "ppc64le"]:
+        DIRAC.gLogger.error(f"Platform {plat} not supported")
+        return None
+
+    CVMFS_locations = DIRAC.gConfig.getValue(
+        "LocalSite/CVMFS_locations", Operations().getValue("Pilot/CVMFS_locations", [])
```

Comment:
What is the point of `CVMFS_locations` here?

Reply:
These locations can be VO dependant.

Reply:
How is that helpful in this context?

Reply:
A VO might want to `/cvmfs/unpacked.cern.ch` and another `cvmfs/dirac.egi.eu`, and so on...
In any case there is always a default.

Reply:
But `continer_root` will basically never make sense on another CVMFS location...

Even if it did, it would be really weird for a VO to set up something different at the same path to try and change the behavior.

Reply:
Ping @fstagni

Reply:
`container_root` IMHO should make sense on any CVMFS location.